tests/common/util.rs

```diff
@@ -72,38 +74,234 @@ pub struct CmdResult {
+        )
+    }
+
+    pub fn stdout_str_apply<'a, F, R>(&'a self, function: F) -> Self
+    where
+        F: Fn(&'a str) -> R,
+        R: Into<Vec<u8>>,
+    {
+        Self::new(
+            self.bin_path.clone(),
+            self.util_name.clone(),
+            self.tmpd.clone(),
+            self.exit_status,
+            function(self.stdout_str()),
+            self.stderr.as_slice(),
+        )
+    }
+
+    pub fn stderr_apply<'a, F, R>(&'a self, function: F) -> Self
+    where
+        F: Fn(&'a [u8]) -> R,
+        R: Into<Vec<u8>>,
+    {
+        Self::new(
+            self.bin_path.clone(),
+            self.util_name.clone(),
+            self.tmpd.clone(),
+            self.exit_status,
+            self.stdout.as_slice(),
+            function(&self.stderr),
+        )
+    }
+
+    pub fn stderr_str_apply<'a, F, R>(&'a self, function: F) -> Self
+    where
+        F: Fn(&'a str) -> R,
+        R: Into<Vec<u8>>,
+    {
+        Self::new(
+            self.bin_path.clone(),
+            self.util_name.clone(),
+            self.tmpd.clone(),
+            self.exit_status,
+            self.stdout.as_slice(),
+            function(self.stderr_str()),
+        )
+    }
+
+    #[track_caller]
+    pub fn stdout_check<'a, F>(&'a self, predicate: F) -> &Self
+    where
+        F: Fn(&'a [u8]) -> bool,
+    {
+        assert!(
+            predicate(&self.stdout),
+            "Predicate for stdout as `bytes` evaluated to false.\nstdout='{:?}'\nstderr='{:?}'\n",
+            &self.stdout,
+            &self.stderr
+        );
+        self
+    }
+
+    #[track_caller]
+    pub fn stdout_str_check<'a, F>(&'a self, predicate: F) -> &Self
+    where
+        F: Fn(&'a str) -> bool,
+    {
+        assert!(
+            predicate(self.stdout_str()),
+            "Predicate for stdout as `str` evaluated to false.\nstdout='{}'\nstderr='{}'\n",
+            self.stdout_str(),
+            self.stderr_str()
+        );
+        self
+    }
+
+    #[track_caller]
+    pub fn stderr_check<'a, F>(&'a self, predicate: F) -> &Self
+    where
+        F: Fn(&'a [u8]) -> bool,
+    {
+        assert!(
+            predicate(&self.stderr),
+            "Predicate for stderr as `bytes` evaluated to false.\nstdout='{:?}'\nstderr='{:?}'\n",
+            &self.stdout,
+            &self.stderr
+        );
+        self
+    }
+
+    #[track_caller]
+    pub fn stderr_str_check<'a, F>(&'a self, predicate: F) -> &Self
+    where
+        F: Fn(&'a str) -> bool,
+    {
+        assert!(
+            predicate(self.stderr_str()),
+            "Predicate for stderr as `str` evaluated to false.\nstdout='{}'\nstderr='{}'\n",
+            self.stdout_str(),
+            self.stderr_str()
+        );
+        self
+    }
+
+    /// Return the exit status of the child process, if any.
+    ///
+    /// Returns None if the child process is still running or hasn't been started.
+    pub fn maybe_exit_status(&self) -> Option<ExitStatus> {
```

Comment:
Nit: I think `try_exit_status` is idiomatic Rust

Reply:
dunno, just tried something here. I searched the `stdlib` once and found no clear answer on how a naming scheme should look like for methods that return an `Option`. `Try` is definitely a good choice for methods that return a `Result`. Most of the time methods that return an `Option` are not given a special prefix, seldomly they are prefixed with `try` or even `maybe`. Sometimes they prefix variables that contain an option with `maybe`. If you'd agree, in our case it's probably best to remove the `maybe` prefix, since this is just a getter and we don't really `try` to get something.

Reply:
I overlooked, that I named another method `exit_status` already. It's `try_exit_status`, then, if that's preferred over `maybe_exit_status`.